SingularityUI/app/components/machines/Slaves.jsx

```diff
@@ -71,7 +72,48 @@ const Slaves = (props) => {
 
   const expiringSlaveState = (slave) => (
     _.find(props.expiringSlaveStates, (expiring) => expiring.machineId == slave.id)
-  )
+  );
+
+  const getMaybeDeactivateHostButton = (slave) => (
+    ! Utils.isIn(slave.host, props.inactiveHosts) && (
+      <FormModalButton
+        name="Mark Inactive"
+        buttonChildren={<Glyphicon glyph="remove-circle" />}
+        action="Mark Host Inactive"
+        onConfirm={() => props.deactivateHost(slave.host)}
+        tooltipText={`Flag host '${slave.host}' as inactive`}
+        formElements={[]}
+      >
+        <p>Are you sure you want to mark the host {slave.host} as inactive?</p>
+        <p>
+          This will decommission every slave on this host until you reactivate
```

Comment:
For clarity, maybe something like:
```
This will automatically decommission any host that joins with a matching hostname.
```
I don't think we will need to mention the piece about reactivation, since the slave will still appear in the decommissioned list with the reactivate button next to it